kerboscript_tests/functest31.ks

```diff
@@ -0,0 +1,38 @@
+//
+// I am not too sure if this is a feature we want to advertise yet,
+// but it does seem to be doable:
+//
+print "Testing varying args logic".
+
+// A function that accepts varying args.
+// the first arg is a number: how many more args to expect.
+// the rest of the args are just printed for testing.
+function print_var_args {
+  parameter how_many.
+  
+  // read parameters, filling a local string.
+  local my_string is "".
+  local i is 1.
+  until i > how_many {
+    parameter next_arg.
+    set my_string to my_string + " " + next_arg.
+    set i to i + 1.
+  }
```

Comment:
Note, given the way the compiler walks the instruction_block body of the function, THIS is the point at which it will insert the opcode to pop the argbottom and assert that it was atop the stack, because it only inserts that check at the outermost level, after the last parameter statement it found.  Because the parameter statement was found inside an `until` loop's braces, the check doesn't happen until those braces end, down here.

This is not a bug.  It's deliberate design, such that if you put a `parameter` statement inside a conditional if block, and the conditional if block doesn't happen, that doesn't cause it to skip over the OpcodeArgBottom.  The OpcodeArgBottom has to occur outside of anything conditional, so it always is guaranteed to happen.